pkg/config/spec.go

```diff
@@ -1025,3 +1015,28 @@ func DetectPreConfiguredDevice(logger sdkTypes.KairosLogger) (string, error) {
 
 	return "", nil
 }
+
+// findUpgradeEntryFromConfig checks the passed config for one of the following ways
+// to set the upgrade entry.
+// - One way is to use the cli arg "--recovery" which makes config.upgrade.entry be "recovery"
+// - Another way is by setting the cli arg "boot-entry" which set it to the specified value.
+// - Lastly, user can set "upgrade.recovery: true" in the kairos config, which should result in entry being "recovery".
+func findUpgradeEntryFromConfig(cfg *Config) string {
+	if _, ok := cfg.Config["upgrade"]; !ok {
+		return ""
```

Comment:
I'm not sure how empty values are handled but that's what the code did up to now so no change here